ingestion_tools/scripts/common/finders.py

```diff
@@ -105,12 +108,47 @@ class BaseLiteralValueFinder(BaseFinder):
     def __init__(self, value: dict[str, str | None] | list[str]):
         self.literal_value = value
 
-    def find(self, config: DepositionImportConfig, glob_vars: dict[str, Any]) -> dict[str, str | None]:
+    def find(self, config: DepositionImportConfig, glob_vars: dict[str, Any], *args, **kwargs) -> dict[str, str | None]:
         if isinstance(self.literal_value, dict):
             return self.literal_value
         return {item: None for item in self.literal_value}
 
 
+class DestinationFilteredMetadataFinder(BaseFinder):
+    # Don't allow reimport the destination files that have already been imported
+    allow_imports: bool = False
+
+    def __init__(self, filters: list[dict[str, Any]]):
+        self.filters = {f.get("key"): f.get("value") for f in filters}
+
+    @classmethod
+    def _is_match(cls, metadata: dict[str, Any], key: str | list[str], expected_value: Any) -> bool:
+        key = [key]
+        value = metadata
+        for path_part in key:
+            if isinstance(value, dict):
+                value = value.get(path_part)
+            else:
+                value = {v.get(path_part) for v in value if isinstance(v, dict)}
+            if not value:
+                break
+
+        return expected_value in value if isinstance(value, set) else value == expected_value
+
+    def find(self, config: DepositionImportConfig, glob_vars: dict[str, Any], *args, **kwargs) -> dict[str, str | None]:
```

Comment:
Can `cls` be a required keyword parameter? I'm actually wondering if the importer class should be a required parameter for the finder constructor instead of a parameter on `find`